src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -12042,6 +12042,12 @@ let TcAndPublishValSpec (cenv: cenv, env, containerInfo: ContainerInfo, declKind
     let attrs = TcAttributes cenv env attrTgt synAttrs
     let newOk = if canInferTypars then NewTyparsOK else NoNewTypars
     let hasNoCompilerInliningAttribute = HasFSharpAttribute g g.attrib_NoCompilerInliningAttribute attrs
+    let hasMethodImplNoInlining = 
+            match TryFindFSharpAttribute g g.attrib_MethodImplAttribute attrs with
+            // NO_INLINING = 8
+            | Some (Attrib(_, _, [ AttribInt32Arg flags ], _, _, _, _)) -> (flags &&& 0x8) <> 0x0
+            | _ -> false
+    let forceDoNotInline = hasNoCompilerInliningAttribute || hasMethodImplNoInlining
```

Comment:
I'm not entirely sure it's correct or am I missing something?

`attrib_MethodImplAttribute` is a hint for JIT only, I am not entirely sure we need to be looking at it.

Presense of `MethodImplAttribute.NoInline` does not mean F# compiler shouldn't inline it, does it?